cmd/rekor-server/app/serve.go

```diff
@@ -16,38 +16,130 @@
+		case "kms":
+			kmshash := viper.GetString("signer.kmshash")
+			hashAlg, ok := hashAlgMap[kmshash]
+			if !ok {
+				slog.Error(fmt.Sprintf("invalid hash algorithm for --signer.kmshash: %s", kmshash))
+				os.Exit(1)
+			}
+			signerOpts = []signer.Option{signer.WithKMS(viper.GetString("signer.kmskey"), hashAlg)}
+		case "tink":
+			signerOpts = []signer.Option{signer.WithTink(viper.GetString("signer.tink_kek_uri"), viper.GetString("signer.tink_keyset_path"))}
+		default:
+			slog.Error(fmt.Sprintf("unrecognized value for --signer.type: %s", signerType))
+			os.Exit(1)
+		}
+		signer, err := signer.New(ctx, signerOpts...)
+		if err != nil {
+			slog.Error(fmt.Sprintf("failed to initialize signer: %v", err.Error()))
+			os.Exit(1)
+		}
+		tesseraStorage, err := tessera.NewStorage(ctx, viper.GetString("hostname"), tesseraDriver, signer)
+		if err != nil {
+			slog.Error(fmt.Sprintf("failed to initialize tessera storage: %v", err.Error()))
+			os.Exit(1)
+		}
+
 		server.Serve(
-			context.Background(),
-			// TODO: use defaults for now, but really use commandline
-			server.NewHTTPConfig(),
-			server.NewGRPCConfig(),
-			&server.Server{},
+			ctx,
+			server.NewHTTPConfig(
+				server.WithHTTPPort(viper.GetInt("http.port")),
+				server.WithHTTPHost(viper.GetString("http.address"))),
+			server.NewGRPCConfig(
+				server.WithGRPCPort(viper.GetInt("grpc.port")),
+				server.WithGRPCHost(viper.GetString("grpc.address"))),
+			server.NewServer(tesseraStorage),
 		)
 	},
 }
 
 func init() {
+	// server configs
+	serveCmd.Flags().Int("http.port", 3000, "HTTP port to bind to")
+	serveCmd.Flags().String("http.address", "127.0.0.1", "HTTP address to bind to")
+	serveCmd.Flags().Int("grpc.port", 3001, "GRPC port to bind to")
+	serveCmd.Flags().String("grpc.address", "127.0.0.1", "GRPC address to bind to")
+	hostname, err := os.Hostname()
+	if err != nil {
+		hostname = "localhost"
+	}
+	serveCmd.Flags().String("hostname", hostname, "public hostname of instance")
+
+	// gcp configs
+	serveCmd.Flags().String("gcp.bucket", "", "GCS bucket for tile and checkpoint storage")
+	serveCmd.Flags().String("gcp.spanner", "", "Spanner database URI")
+
+	// checkpoint signing configs
+	serveCmd.Flags().String("signer.type", "file", "type of signer to use, must be one of 'file', 'kms', or 'tink'")
+	serveCmd.Flags().String("signer.filepath", "", "path to the signing key")
+	serveCmd.Flags().String("signer.password", "", "password to decrypt the signing key")
+	serveCmd.Flags().String("signer.kmskey", "", "URI of the KMS key, in the form of awskms://keyname, azurekms://keyname, gcpkms://keyname, or hashivault://keyname")
+	serveCmd.Flags().String("signer.kmshash", "sha256", "hash algorithm used by the KMS")
+	serveCmd.Flags().String("signer.tink_kek_uri", "", "encryption key for decrypting Tink keyset. Valid options are [aws-kms://keyname, gcp-kms://keyname]")
```

Comment:
nit, preference for hyphenated CLI flags, following https://www.gnu.org/software/libc/manual/html_node/Argument-Syntax.html